thevoid/connection.cpp

```diff
@@ -23,6 +23,70 @@
 #include "stream_p.hpp"
 #include "stockreplies_p.hpp"
 
+namespace {
+
+const long int USECS_IN_SEC = 1000 * 1000;
+const long int NSECS_IN_SEC = USECS_IN_SEC * 1000;
+
+struct timespec& operator+= (struct timespec& lhs, const struct timespec& rhs) {
+	lhs.tv_sec += rhs.tv_sec;
+	lhs.tv_nsec += rhs.tv_nsec;
+
+	if (lhs.tv_nsec >= NSECS_IN_SEC) {
+		lhs.tv_sec += 1;
+		lhs.tv_nsec -= NSECS_IN_SEC;
+	}
+
+	return lhs;
+}
+
+struct timespec operator+ (const struct timespec& lhs, const struct timespec& rhs) {
+	struct timespec res(lhs);
+	res += rhs;
+	return res;
+}
```

Comment:
The only thing lhs, that is passed by constant lvalue reference to avoid copy, is used for is to be copied into res. 
Why lhs shouldn't be passed by value? In this case the operator's body is just: ```return lhs += rhs;```